modules/deploy/pages/deployment-option/cloud/aws-privatelink.adoc

```diff
@@ -0,0 +1,394 @@
++
+--
+- `allowed_principals`: Amazon Resource Names (ARNs) for the AWS principals allowed to access the endpoint service. For example, for all principals in an account, use `arn:aws:iam::account_id:root`. See https://docs.aws.amazon.com/vpc/latest/privatelink/configure-endpoint-service.html#add-remove-permission[Configure an endpoint service] for details.
+--
++
+[,bash]
+----
+CLUSTER_PATCH_BODY=`cat << EOF
+{
+  "private_link": {
+     "enabled": true,
+     "aws": {
+        "allowed_principals": ["<principal_1>","<principal_2>", ...]
+     }
+  }
+}
+EOF`
+
+curl -vv -X PATCH \
+  -H "Content-Type: application/json" \
+  -H "Authorization: Bearer $AUTH_TOKEN" \
+  -d "$CLUSTER_PATCH_BODY" $PUBLIC_API_ENDPOINT/v1beta1/clusters/$CLUSTER_ID
+----
+
+. Before proceeding, check the state of the Update Cluster operation by calling https://docs.api.redpanda.com/#get-/v1beta1/operations/-id-[GET /v1beta1/operations/{id}], passing the operation ID returned from Update Cluster call. When the state is `STATE_READY`, proceed to the next step.
+
+. Check the service state by calling https://docs.api.redpanda.com/#get-/v1beta1/clusters/-id-[`GET /v1beta1/clusters/{id}`]. The `service_state` in the `private_link.status` response object needs to be `Available` for you to <<connect-to-privatelink-endpoint-service,connect to the service>>.
++
+[,bash]
+----
+curl -X GET \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $AUTH_TOKEN" \
+    $PUBLIC_API_ENDPOINT/v1beta1/clusters/$CLUSTER_ID | jq '.private_link.status.aws | {service_name, service_state}'
+----
+
+== Connect to PrivateLink endpoint service
+
+=== Get cluster domain
+
+Get the domain (`cluster_domain`) of the cluster from the cluster details in the Redpanda Cloud UI.
+
+For example, if the Bootstrap server URL is: `seed-3da65a4a.cki01qgth38kk81ard3g.fmc.dev.cloud.redpanda.com:9092`, then `cluster_domain` is: `cki01qgth38kk81ard3g.fmc.dev.cloud.redpanda.com`.
+
+[,bash]
+----
+CLUSTER_DOMAIN=<cluster_domain>
+----
+
+=== Get name of PrivateLink endpoint service
+
+You need the service name to <<create-vpc-endpoint,create VPC private endpoints>>. Run the following command to get the service name:
+
+[,bash]
+----
+PL_SERVICE_NAME=`curl -X GET \
+  -H "Content-Type: application/json" \
+  -H "Authorization: Bearer $AUTH_TOKEN" \
+  $PUBLIC_API_ENDPOINT/v1beta1/clusters/$CLUSTER_ID | jq -r .private_link.status.aws.service_name`
+----
+
+=== Create client VPC
+
+If you are not using an existing VPC, you must create a new one. 
+
+The VPC region must be the same region where the Redpanda cluster is deployed. Run the following command to create the VPC:
+
+[,bash]
+----
+PROFILE=<specific-profile-from-your-credential-file>
+
+aws ec2 create-vpc --region $REGION --profile $PROFILE --cidr-block 10.0.0.0/20
+CLIENT_VPC_ID=<client_vpc_id>
+----
+
+You can also use an existing VPC. You need the VPC ID to <<modify-vpc-attributes,modify its attributes>>.
+
+=== Modify VPC attributes
```

Comment:
@paulzhang97 just to double check, which VPC attributes are being modified here? Is it the --enable-dns-hostnames and/or --enable-dns-support flag?